homeassistant/components/ruckus_unleashed/device_tracker.py

```diff
@@ -0,0 +1,117 @@
+"""Support for Ruckus Unleashed devices."""
+from homeassistant.components.device_tracker import (
+    ATTR_MAC,
+    ATTR_SOURCE_TYPE,
+    SOURCE_TYPE_ROUTER,
+)
+from homeassistant.components.device_tracker.config_entry import ScannerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers import entity_registry
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CLIENTS, COORDINATOR, DOMAIN, UNDO_UPDATE_LISTENERS
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for Ruckus Unleashed component."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+
+    tracked = set()
+
+    @callback
+    def router_update():
+        """Update the values of the router."""
+        add_new_entities(coordinator, async_add_entities, tracked)
+
+    router_update()
+
+    hass.data[DOMAIN][entry.entry_id][UNDO_UPDATE_LISTENERS].append(
+        coordinator.async_add_listener(router_update)
+    )
+
+    registry = await entity_registry.async_get_registry(hass)
+    restore_entities(registry, coordinator, entry, async_add_entities, tracked)
+
+
+@callback
+def add_new_entities(coordinator, async_add_entities, tracked):
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    for mac in coordinator.data[CLIENTS].keys():
+        if mac in tracked:
+            continue
+
+        device = coordinator.data[CLIENTS][mac]
+        new_tracked.append(RuckusUnleashedDevice(coordinator, mac, device[CONF_NAME]))
+        tracked.add(mac)
+
+    if new_tracked:
+        async_add_entities(new_tracked, True)
```

Comment:
Do we need to request a new fetch of device data?

Reply:
No, by this point all of the new data has been loaded by the `DataUpdateCoordinator`

Reply:
Good! Then we can remove the second argument True.

Reply:
Thank you, good catch. That was leftover from before I had a `DataUpdateCoordinator`